### PR TITLE
Jsf22 apar tests jakarta ee10 and MYFACES-4540

### PR DESCRIPTION
**Pull request by pnicolucci**
for #22369

Pulling in https://issues.apache.org/jira/browse/MYFACES-4540 to solve the Java2 Security issue.

### CODE REVIEW

**Comment by pnicolucci** [comment]
#libby #build

**Comment by pnicolucci** [comment]
Java 17 build: https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.build.viewResult&id=_wtCkoHAOEe2SBN8svFwMrQ

**Comment by pnicolucci** [comment]
Java 17 build was successful

**Comment by pnicolucci** [comment]
#build

**Review by volosied — DISMISSED** [review]
Looks good to me.

**Comment by pnicolucci** [comment]
#libby #build

File: dev/com.ibm.ws.jsf.2.2_fat/fat/src/com/ibm/ws/jsf22/fat/tests/JSF22AparTests.java [hunk 1]
```diff
@@ -886,7 +906,9 @@ public void testPI89168() throws Exception {
 
             // Check that a FacesException is thrown
             assertTrue("PI89168: FacesException was not thrown!\n" + page.asText(),
-                       page.asText().contains((JakartaEE9Action.isActive() ? "jakarta." : "javax.") + "faces.FacesException: unable to create view \"/statefulView.xhtml\""));
+                       page.asText()
+                                       .contains((JakartaEE10Action.isActive() || JakartaEE9Action.isActive() ? "jakarta." : "javax.")
```
Comment (volosied) [inline dev/com.ibm.ws.jsf.2.2_fat/fat/src/com/ibm/ws/jsf22/fat/tests/JSF22AparTests.java:910]: No need for JakartaEE10Action.isActive() if EE10 is skipped.

File: dev/io.openliberty.org.apache.myfaces.4.0/src/org/apache/myfaces/util/WebXmlParser.java [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+
+            if (System.getSecurityManager() != null)
+            {
+                is = (InputStream) AccessController.doPrivileged(
+                        (PrivilegedExceptionAction<Object>) () -> url.openStream());
+            }
+            else
+            {
+                is = url.openStream();
+            }
```
Comment (volosied) [inline dev/io.openliberty.org.apache.myfaces.4.0/src/org/apache/myfaces/util/WebXmlParser.java]: Is there a JIRA for this yet? I can't recall.

Comment (pnicolucci) [reply dev/io.openliberty.org.apache.myfaces.4.0/src/org/apache/myfaces/util/WebXmlParser.java]: No, not yet wanted to get a review before opening a JIRA. I'll do that next and update the commit messages so it is clear what JIRA we're overlaying here. I won't merge until it is accepted in MyFaces

**Review by volosied — DISMISSED** [review]

**Comment by pnicolucci** [comment]
#libby #build #spawn.fullfat.buckets="com.ibm.ws.jsf.2.2_fat"

**Comment by pnicolucci** [comment]
#libby #build

File: dev/io.openliberty.org.apache.myfaces.4.0/src/org/apache/myfaces/util/WebXmlParser.java (hunk 2 above)
Comment (pnicolucci) [reply dev/io.openliberty.org.apache.myfaces.4.0/src/org/apache/myfaces/util/WebXmlParser.java]: https://issues.apache.org/jira/browse/MYFACES-4540

**Comment by pnicolucci** [comment]
#libby #build

**Review by volosied — APPROVED** [review]
Thanks!

**Pull request merged**